Automattic/jetpack
# Payments: Remove in-editor Stripe nudge

Fixes 178-gh-Automattic/view-design

#### Changes proposed in this Pull Request:

* Removes the old in-editor Stripe nudge on the Payments block, and adds a Connect Stripe button in the block toolbar. Similar to the work done for Premium Content.

Old version:
<img width="635" alt="Screen Shot 2021-01-13 at 1 37 36 PM" src="https://user-images.githubusercontent.com/63313398/104513454-8c31bf00-55a4-11eb-9d91-53bc92ec78b6.png">

New version:
<img width="407" alt="Screen Shot 2021-01-13 at 12 59 52 PM" src="https://user-images.githubusercontent.com/63313398/104513482-9653bd80-55a4-11eb-80e3-ad1b99b79124.png">


#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






Use https://github.com/Automattic/a8c-wp-env to test.
* Start up a site with `npm run start`
* Check out this branch and activate jurassic.tube with npm run tube:init. 
* Activate jetpack on the new site but do not add a plan yet.
* Create a new post and insert the Payments block.
* Verify that there is an Upgrade banner, but the "Connect Stripe" button is not available in the block toolbar
* Add a "Security Daily" Jetpack plan to the site
* Go back to the post. Confirm that the upgrade banner has been removed, and the new "Connect Stripe" button is in the block toolbar (matching screenshot above)
* Click on the "Connect Stripe" button and verify that you are able to connect to a Stripe account (use the ryan.ray test stripe account in the secret store and confirm that the connection works)
* Confirm that the "Connect Stripe" button is now missing from the toolbar.


#### Proposed changelog entry for your changes:


*
